src/components/percentage/percentage.jsx

```diff
@@ -0,0 +1,148 @@
+import PropTypes from 'prop-types';
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+} from 'react';
+import FormControl from '../form-control/form-control.jsx';
+import FormControlIcons from '../form-control-icons/form-control-icons.jsx';
+import styles from './percentage.css';
+import useMounted from '../../hooks/use-mounted.js';
+import useValidation from '../../hooks/use-validation.jsx';
+
+function getClassName(className, validationMessage) {
+  if (className) return className;
+  return validationMessage ? styles['invalid-input'] : styles.input;
+}
+
+const Percentage = forwardRef(function Percentage(props, forwardedRef) {
+  const fallbackRef = useRef();
+  const ref = forwardedRef || fallbackRef;
+  const inputRef = useRef();
+  const mounted = useMounted();
+  const [validationMessage, validate] = useValidation(props.validationMessage, inputRef);
+
+  function onBlur(event) {
+    validate();
+    props.onBlur(event);
+  }
+
+  function onChange(event) {
+    validate();
+    props.onChange(event);
+  }
+
+  useEffect(() => {
+    if (!mounted.current) return;
+
+    // The MDS Input is using an uncontrolled `<input>`.
+    // In order to set a new provided value prop, we
+    // set the internal state of the `<input>`.
+    inputRef.current.value = props.value || '';
+  }, [props.value]);
+
+  useImperativeHandle(ref, () => ({
+    get dirty() {
+      const providedValue = props.value || undefined;
+      return !Object.is(providedValue, this.value);
+    },
+    id: props.id,
+    name: props.name,
+    get value() {
+      return parseFloat(inputRef.current.value) || undefined;
```

Comment:
In case ya wanna get in the habit of referencing the global instead its API. 

```suggestion
       return window.parseFloat(inputRef.current.value) || undefined;
```